### ISSUE TITLE
Factor creation is not semiring safe

### ISSUE THREAD

**Issue by bruttenberg** [p2t2/figaro#656]
In the Factor creation code in Factory, several methods hard code 1.0 or 0.0 into the factors. These values should be tied to the semiring type.

**Comment by wkretschmer** [comment]
Isn't this because all of the factor creation methods assume SumProductSemiring? Otherwise, you would have to e.g. know to convert the probabilities in a Select to log space. Are there even any factory methods that let you specify a semiring?

**Comment by bruttenberg** [comment]
You’re right, but I’m a little worried about SFI, since factors are created while the algorithm is running, I’m not sure if we can guarantee that all the necessary conversion will be done.

+  +  +  +  +  +  +  +  +  +  +  +  +  +
Brian Ruttenberg, PhD
Senior Scientist
Decision Management Systems Division
Government Services
Charles River Analytics Inc.
617.491.3474 x730
www.cra.com<http://www.cra.com/>
